### ISSUE TITLE
make onUpgrade() more modular so we don't have to re-download all data on every upgrade

### ISSUE THREAD

**Issue by jakewilson** [dharmaseed/dharmaseed-android#62]
We should have things to do for specific upgrades so we don't have to drop every table (which is what we do now) and re-download it. For example, adding a new column to the centers table shouldn't cause us to drop the Talk table and take 10 minutes re-downloading it.

**Comment by jakewilson** [comment]
@bb4242 do you know how many people have downloaded the app? I wonder if, before we officially release the app, we could do a clean restart and set the `DB_VERSION` to 1? Then we could get rid of all logic in `onUpgrade()` and from then on we could make it more modular from the ground up, whenever we need to update it. I don't know if this is a terrible idea though. What do you think?

**Comment by bb4242** [comment]
Yeah, I think that's a great idea.  Haha, the total number of installs right now is 2 (one of them is me, and I guess probably @ewoudenberg also; the app is not yet available to the public), so I'd say resetting before releasing to the public would be a great time to do it.
